back-end/src/models/AppUser/AppUser.repository.ts

```diff
@@ -163,6 +184,61 @@ export default class AppUserRepository extends AppUserDb {
     });
   }
 
+  static async updateUserPasswordWithToken(
+    token: string,
+    password: string
+  ): Promise<AppUser> {
+    // Check if token is valid
+    const userToUpdate = await this.getUserByToken(token);
+    if (!userToUpdate) {
+      throw new Error("Aucun utilisateur ne correspond à ce token.");
+    }
+
+    // Check if token is expired
+    const resetPasswordTokenExpiration =
+      userToUpdate.resetPasswordTokenExpiration;
+    if (!resetPasswordTokenExpiration) {
+      throw new Error("Ce token n'est pas valide.");
+    }
+    if (resetPasswordTokenExpiration < new Date()) {
+      throw new Error("Ce token a expiré.");
+    }
+
+    // Reset expiration date
+    const newResetPasswordTokenExpiration = new Date();
+
+    // Update user updatedAt
+    const updatedAt = new Date();
+
+    // Save user
+    return this.repository.save({
+      id: userToUpdate.id,
+      hashedPassword: hashSync(password),
+      updatedAt: updatedAt,
+      resetPasswordTokenExpiration: newResetPasswordTokenExpiration,
+    });
+  }
+
+  static async updateUserToken(
+    id: string,
+    resetPasswordToken: string
+  ): Promise<AppUser> {
+    const userToUpdate = await this.getUserById(id);
+
+    if (!userToUpdate) {
+      throw new Error("Aucun utilisateur ne correspond à cet ID.");
+    }
+
+    // Token expiration date set to 30 minutes
+    const resetPasswordTokenExpiration = new Date(Date.now() + 1000 * 60 * 30);
```

Comment:
You can replace it by : 
`const resetPasswordTokenExpiration = DateUpdates.addMinutesToDate(new Date(), 30);`

Reply:
ok

Reply:
Fix after VD codereview for the backend is ok